core/rawdb/accessors_chain.go

```diff
@@ -708,10 +708,12 @@ func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
 // DeriveLogFields fills the logs in receiptLogs with information such as block number, txhash, etc.
 func deriveLogFields(receipts []*receiptLogs, hash common.Hash, number uint64, txs types.Transactions) error {
 	logIndex := uint(0)
-	if len(txs) != len(receipts) {
+	// The receipts may include an additional "block finalization" receipt (only IBFT)
+	if !(len(txs) == len(receipts) || len(txs)+1 == len(receipts)) {
 		return errors.New("transaction and receipt count mismatch")
 	}
-	for i := 0; i < len(receipts); i++ {
+	// len(r) is not always strictly equal to len(txs) because of the block finalization receipt (IBFT)
```

Comment:
clarify in the comment that the block finalization receipt is the last one from the array